drivers/scaleway/scaleway.go

```diff
@@ -0,0 +1,376 @@
+		return "", fmt.Errorf("IP address is not set")
+	}
+	return d.IPAddress, nil
+}
+
+func (d *Driver) GetState() (state.State, error) {
+	log.Debugf("[Scaleway] Retrieving state server %s", d.Id)
+	client := d.getClient()
+	response, err := client.GetServer(d.Id)
+	if err != nil {
+		return state.Error, err
+	}
+	return getServerState(response.Server.State), nil
+}
+
+func (d *Driver) Start() error {
+	log.Infof("[Scaleway] Starting instance...")
+	client := d.getClient()
+	if _, err := client.PerformServerAction(d.Id, "poweron"); err != nil {
+		return err
+	}
+	d.waitForServerState(state.Running)
+	return nil
+}
+
+func (d *Driver) Stop() error {
+	log.Infof("[Scaleway] Stopping instance...")
+	client := d.getClient()
+	if _, err := client.PerformServerAction(d.Id, "poweroff"); err != nil {
+		return err
+	}
+	d.waitForServerState(state.Stopped)
+	return nil
+}
+
+func (d *Driver) Remove() error {
+	log.Infof("[Scaleway] Removing instance... ")
+	client := d.getClient()
+	if err := client.DeleteServer(d.Id); err != nil {
+		return err
+	}
+	d.waitForServerState(state.Stopped)
+	return nil
+}
+
+func (d *Driver) Restart() error {
+	log.Infof("[Scaleway] Rebooting instance...")
+	client := d.getClient()
+	if _, err := client.PerformServerAction(d.Id, "reboot"); err != nil {
+		return err
+	}
+	d.waitForServerState(state.Running)
+	return nil
+}
+
+func (d *Driver) Kill() error {
+	return d.Stop()
+}
+
+// func (d *Driver) Upgrade() error {
+// 	log.Debugf("[Scaleway] Upgrading Docker")
+// 	_, err := drivers.RunSSHCommandFromDriver(d,
+// 		"sudo apt-get update && apt-get install --upgrade lxc-docker")
+// 	return err
+// }
+
+func (d *Driver) setupHostname() error {
+	log.Debugf("[Scaleway] Setting hostname: %s", d.MachineName)
+	_, err := drivers.RunSSHCommandFromDriver(d,
+		fmt.Sprintf(
+			"echo \"127.0.0.1 %s\" | sudo tee -a /etc/hosts && sudo hostname %s && echo \"%s\" | sudo tee /etc/hostname",
+			d.MachineName,
+			d.MachineName,
+			d.MachineName,
+		))
+	return err
+}
+
+// func (d *Driver) installDocker() error {
+// 	_, err := drivers.RunSSHCommandFromDriver(d,
+// 		"if [ ! -e /usr/bin/docker ]; then curl -sL https://get.docker.com | sh -; fi")
+// 	return err
+// }
+
+// func (d *Driver) StartDocker() error {
+// 	log.Debug("Starting Docker...")
+// 	_, err := drivers.RunSSHCommandFromDriver(d,
+// 		"sudo service docker start")
+// 	return err
+// }
+
+// func (d *Driver) StopDocker() error {
+// 	log.Debug("Stopping Docker...")
+// 	_, err := drivers.RunSSHCommandFromDriver(d,
+// 		"sudo service docker stop")
+// 	return err
+// }
+
+func (d *Driver) setMachineNameIfNotSet() {
+}
+
+func (d *Driver) createSSHKey() error {
+	if err := ssh.GenerateSSHKey(d.GetSSHKeyPath()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Driver) publicSSHKeyPath() string {
+	return d.GetSSHKeyPath() + ".pub"
+}
+
+func (d *Driver) getClient() *api.ScalewayClient {
+	return api.NewClient(d.UserId, d.Token, d.Organization)
```

Comment:
The order of arguments does not match https://github.com/nlamirault/go-scaleway/blob/master/api/client.go#L48